Clarify generate-compile-commands is only needed for XCode builds

ntrrgc opened the pull request:
#### 2a4191b4fda114f2781cb0ae27eb7571d2ecffa0
<pre>
Clarify generate-compile-commands is only needed for XCode builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=271053">https://bugs.webkit.org/show_bug.cgi?id=271053</a>

Reviewed by Darin Adler.

This patch adds clarifications that generate-compile-commands is only
needed for XCode based builds (i.e. current Mac builds).

An additional hint is added to the script such that if it detects the
build is CMake based it instructs the user on how to generate
compile_commands.json with CMake.

* Tools/Scripts/build-webkit:
* Tools/Scripts/generate-compile-commands:

Canonical link: <a href="https://commits.webkit.org/277197@main">https://commits.webkit.org/277197@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1134f4341907db0335f141dd68e6577dd8c1ec85

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44409 "Build is in progress. Recent messages:OS: Unknown (10.15.7), Xcode: 11.7") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23483 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46860 "Build is in progress. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47178 "Built successfully") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/WinCairo-EWS "Waiting in queue, processing has not started yet") 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46715 "Build is in progress. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27476 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20878 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36618 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44986 "Build is in progress. Recent messages:OS: Unknown (10.15.7), Xcode: 11.7") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/27476 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38232 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17682 "Passed tests") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/27476 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39356 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2460 "Build is in progress. Recent messages:Printed configuration") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/37653 "Built successfully and passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/27476 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39637 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48678 "Build is in progress. Recent messages:Printed configuration") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43905 "Built successfully and passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19390 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15923 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/48678 "Build is in progress. Recent messages:Printed configuration") | 
| | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/46860 "Build is in progress. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/48678 "Build is in progress. Recent messages:Printed configuration") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10370 "Built successfully and passed tests") | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/51047 "Build is in progress. Recent messages:Printed configuration") | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/35/builds/51047 "Build is in progress. Recent messages:Printed configuration") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Tools/Scripts/build-webkit:
```diff
@@ -150,7 +150,7 @@ Usage: $programName [options] [options to pass to build system]
   --[no-]use-ccache                 Enable (or disable) CCache, if available
 
-  --export-compile-commands         Generate compile_commands.json (Apple platforms only)
+  --export-compile-commands         Generate compile_commands.json (XCode builds only)
```
Xcode is spelled with a lowercase "c".

darinadler commented on Tools/Scripts/generate-compile-commands:
```diff
@@ -36,7 +36,7 @@ import json
 
 # Get command line args
-parser = argparse.ArgumentParser(description='Generate compile_commands.json', 
+parser = argparse.ArgumentParser(description='Generate compile_commands.json for XCode builds',
```
Xcode is spelled with a lowercase "c".

darinadler commented on Tools/Scripts/generate-compile-commands:
```diff
@@ -57,8 +57,15 @@ os.chdir(base_dir)
+    is_cmake_build = os.path.exists(opts.built_products_dir + "/CMakeCache.txt")
+    if is_cmake_build:
+        print("The build directory passed seems to be a CMake build.")
+        print("This script is only used for XCode builds.")
```
Xcode is spelled with a lowercase "c".

ntrrgc committed:
Clarify generate-compile-commands is only needed for XCode builds
https://bugs.webkit.org/show_bug.cgi?id=271053

Reviewed by Darin Adler.

This patch adds clarifications that generate-compile-commands is only
needed for XCode based builds (i.e. current Mac builds).

An additional hint is added to the script such that if it detects the
build is CMake based it instructs the user on how to generate
compile_commands.json with CMake.

* Tools/Scripts/build-webkit:
* Tools/Scripts/generate-compile-commands:

Canonical link: https://commits.webkit.org/277197@main
